sockeye/inference.py

```diff
@@ -1400,16 +1387,15 @@ def _beam_search(self,
 
         best_word_indices = mx.nd.full((self.batch_size * self.beam_size,), val=self.start_id, ctx=self.context,
                                        dtype='int32')
-        # growing sequences: (batch_size * beam_size, 1), pre-filled with <s> symbols on index 0
-        sequences = mx.nd.full((self.batch_size * self.beam_size, 1), val=self.start_id, ctx=self.context,
-                               dtype='int32')
+        # The growing sequences representing the complete history for each hypothesis: (batch_size * beam_size, 1)
+        sequences = mx.nd.zeros((self.batch_size * self.beam_size, 1), ctx=self.context, dtype='int32')
```

Comment:
I think this obfuscates the fact that the 0th column represents `<s>`. Could you add a comment that the 0th column is stripped/not used in the end anyway?